bug: failed to build Tests for WASI-NN Piper on Ubuntu 20.04

### Summary

Link to CI failure: https://github.com/WasmEdge/WasmEdge/actions/runs/10960134124/job/30434082962

(Manually reproduced) CMakeError.log:
```
Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /root/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_88062 && [1/2] Building C object CMakeFiles/cmTC_88062.dir/src.c.o
[2/2] Linking C executable cmTC_88062
FAILED: cmTC_88062 
: && /usr/bin/clang-12 -DCMAKE_HAVE_LIBC_PTHREAD   CMakeFiles/cmTC_88062.dir/src.c.o  -o cmTC_88062   && :
/usr/bin/ld: CMakeFiles/cmTC_88062.dir/src.c.o: in function `main':
src.c:(.text+0x35): undefined reference to `pthread_create'
/usr/bin/ld: src.c:(.text+0x3e): undefined reference to `pthread_detach'
/usr/bin/ld: src.c:(.text+0x4b): undefined reference to `pthread_join'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.


Source file was:
#include <pthread.h>

void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}
```


### Current State

Currently CI don't have a job for testing WASI-NN Piper on Ubuntu 20.04 (x86_64).

### Expected State

_No response_

### Reproduction steps

Build with docker image: `wasmedge/wasmedge:ubuntu-20.04-build-clang-plugins-deps`
```
cmake -Bbuild -GNinja \
    -DCMAKE_BUILD_TYPE=Release \
    -DWASMEDGE_BUILD_TOOLS=OFF \
    -DWASMEDGE_USE_LLVM=OFF \
    -DWASMEDGE_PLUGIN_WASI_NN_BACKEND=Piper
cmake -Bbuild -DWASMEDGE_BUILD_TESTS=ON                      # fail here
```

NOTE: Warnings of ONNX exist with and without `-DWASMEDGE_BUILD_TESTS=ON`

Log for the second (failed) command:
```
-- Downloading SIMDJSON source
-- Adding -Og to compile flag
-- Downloading SIMDJSON source -- done
-- Downloading the WASM spec test suite
-- Downloading the WASM spec test suite -- done
-- Copying test suite to /root/build/test/spec/testSuites/core
-- Copying test suite to /root/build/test/spec/testSuites/core -- done
-- Copying test suite to /root/build/test/spec/testSuites/multi-memory
-- Copying test suite to /root/build/test/spec/testSuites/multi-memory -- done
-- Copying test suite to /root/build/test/spec/testSuites/tail-call
-- Copying test suite to /root/build/test/spec/testSuites/tail-call -- done
-- Copying test suite to /root/build/test/spec/testSuites/extended-const
-- Copying test suite to /root/build/test/spec/testSuites/extended-const -- done
-- Copying test suite to /root/build/test/spec/testSuites/threads
-- Copying test suite to /root/build/test/spec/testSuites/threads -- done
-- Copying test suite to /root/build/test/spec/testSuites/function-references
-- Copying test suite to /root/build/test/spec/testSuites/function-references -- done
-- Copying test suite to /root/build/test/spec/testSuites/gc
-- Copying test suite to /root/build/test/spec/testSuites/gc -- done
-- Copying test suite to /root/build/test/spec/testSuites/exception-handling
-- Copying test suite to /root/build/test/spec/testSuites/exception-handling -- done
-- Copying test suite to /root/build/test/spec/testSuites/exception-handling-legacy
-- Copying test suite to /root/build/test/spec/testSuites/exception-handling-legacy -- done
-- Copying test suite to /root/build/test/spec/testSuites/relaxed-simd
-- Copying test suite to /root/build/test/spec/testSuites/relaxed-simd -- done
-- Download ML artifacts to /root/build/test/plugins/wasi_nn/wasinn_piper_fixtures
-- [download 0% complete]
-- [download 1% complete]
-- [download 2% complete]
-- [download 3% complete]
-- [download 4% complete]
-- [download 5% complete]
-- [download 6% complete]
-- [download 7% complete]
-- [download 8% complete]
-- [download 9% complete]
-- [download 10% complete]
-- [download 11% complete]
-- [download 12% complete]
-- [download 13% complete]
-- [download 14% complete]
-- [download 15% complete]
-- [download 16% complete]
-- [download 17% complete]
-- [download 18% complete]
-- [download 19% complete]
-- [download 20% complete]
-- [download 21% complete]
-- [download 22% complete]
-- [download 23% complete]
-- [download 24% complete]
-- [download 25% complete]
-- [download 26% complete]
-- [download 27% complete]
-- [download 28% complete]
-- [download 29% complete]
-- [download 30% complete]
-- [download 31% complete]
-- [download 32% complete]
-- [download 33% complete]
-- [download 34% complete]
-- [download 35% complete]
-- [download 36% complete]
-- [download 37% complete]
-- [download 38% complete]
-- [download 39% complete]
-- [download 40% complete]
-- [download 41% complete]
-- [download 42% complete]
-- [download 43% complete]
-- [download 44% complete]
-- [download 45% complete]
-- [download 46% complete]
-- [download 47% complete]
-- [download 48% complete]
-- [download 49% complete]
-- [download 50% complete]
-- [download 51% complete]
-- [download 52% complete]
-- [download 53% complete]
-- [download 54% complete]
-- [download 55% complete]
-- [download 56% complete]
-- [download 57% complete]
-- [download 58% complete]
-- [download 59% complete]
-- [download 60% complete]
-- [download 61% complete]
-- [download 62% complete]
-- [download 63% complete]
-- [download 64% complete]
-- [download 65% complete]
-- [download 66% complete]
-- [download 67% complete]
-- [download 68% complete]
-- [download 69% complete]
-- [download 70% complete]
-- [download 71% complete]
-- [download 72% complete]
-- [download 73% complete]
-- [download 74% complete]
-- [download 75% complete]
-- [download 76% complete]
-- [download 77% complete]
-- [download 78% complete]
-- [download 79% complete]
-- [download 80% complete]
-- [download 81% complete]
-- [download 82% complete]
-- [download 83% complete]
-- [download 84% complete]
-- [download 85% complete]
-- [download 86% complete]
-- [download 87% complete]
-- [download 88% complete]
-- [download 89% complete]
-- [download 90% complete]
-- [download 91% complete]
-- [download 92% complete]
-- [download 93% complete]
-- [download 94% complete]
-- [download 95% complete]
-- [download 96% complete]
-- [download 97% complete]
-- [download 98% complete]
-- [download 99% complete]
-- [download 100% complete]
-- [download 100% complete]
-- [download 0% complete]
-- [download 1% complete]
-- [download 2% complete]
-- [download 3% complete]
-- [download 4% complete]
-- [download 5% complete]
-- [download 6% complete]
-- [download 7% complete]
-- [download 8% complete]
-- [download 9% complete]
-- [download 10% complete]
-- [download 11% complete]
-- [download 12% complete]
-- [download 13% complete]
-- [download 14% complete]
-- [download 15% complete]
-- [download 16% complete]
-- [download 17% complete]
-- [download 18% complete]
-- [download 19% complete]
-- [download 20% complete]
-- [download 21% complete]
-- [download 22% complete]
-- [download 23% complete]
-- [download 24% complete]
-- [download 25% complete]
-- [download 26% complete]
-- [download 27% complete]
-- [download 28% complete]
-- [download 29% complete]
-- [download 30% complete]
-- [download 31% complete]
-- [download 32% complete]
-- [download 33% complete]
-- [download 34% complete]
-- [download 35% complete]
-- [download 36% complete]
-- [download 37% complete]
-- [download 38% complete]
-- [download 39% complete]
-- [download 40% complete]
-- [download 41% complete]
-- [download 42% complete]
-- [download 43% complete]
-- [download 44% complete]
-- [download 45% complete]
-- [download 46% complete]
-- [download 47% complete]
-- [download 48% complete]
-- [download 49% complete]
-- [download 50% complete]
-- [download 51% complete]
-- [download 52% complete]
-- [download 53% complete]
-- [download 54% complete]
-- [download 55% complete]
-- [download 56% complete]
-- [download 57% complete]
-- [download 58% complete]
-- [download 59% complete]
-- [download 60% complete]
-- [download 61% complete]
-- [download 62% complete]
-- [download 63% complete]
-- [download 64% complete]
-- [download 65% complete]
-- [download 66% complete]
-- [download 67% complete]
-- [download 68% complete]
-- [download 69% complete]
-- [download 70% complete]
-- [download 71% complete]
-- [download 72% complete]
-- [download 73% complete]
-- [download 74% complete]
-- [download 75% complete]
-- [download 76% complete]
-- [download 77% complete]
-- [download 78% complete]
-- [download 79% complete]
-- [download 80% complete]
-- [download 81% complete]
-- [download 82% complete]
-- [download 83% complete]
-- [download 84% complete]
-- [download 85% complete]
-- [download 86% complete]
-- [download 87% complete]
-- [download 88% complete]
-- [download 89% complete]
-- [download 90% complete]
-- [download 91% complete]
-- [download 92% complete]
-- [download 93% complete]
-- [download 94% complete]
-- [download 95% complete]
-- [download 96% complete]
-- [download 97% complete]
-- [download 98% complete]
-- [download 99% complete]
-- [download 100% complete]
CMake Error at test/plugins/wasi_nn/CMakeLists.txt:92 (file):
  file does not recognize sub-command ARCHIVE_EXTRACT


-- WASI-NN: Build piper backend for WASI-NN
-- {fmt} version: 11.0.2
-- Build type: Release
-- Build spdlog: 1.13.0
-- Build type: Release
CMake Warning at cmake/WASINNDeps.cmake:238 (find_package):
  By not providing "Findonnxruntime.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "onnxruntime", but CMake did not find one.

  Could not find a package configuration file provided by "onnxruntime" with
  any of the following names:

    onnxruntimeConfig.cmake
    onnxruntime-config.cmake

  Add the installation prefix of "onnxruntime" to CMAKE_PREFIX_PATH or set
  "onnxruntime_DIR" to a directory containing one of the above files.  If
  "onnxruntime" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  cmake/WASINNDeps.cmake:185 (wasmedge_setup_piper)
  test/plugins/wasi_nn/CMakeLists.txt:119 (wasmedge_setup_wasinn_target)


-- Downloading piper source
CMake Warning at /root/build/_deps/piper_phonemize-src/CMakeLists.txt:50 (find_package):
  By not providing "Findonnxruntime.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "onnxruntime", but CMake did not find one.

  Could not find a package configuration file provided by "onnxruntime"
  (requested version 1.14.1) with any of the following names:

    onnxruntimeConfig.cmake
    onnxruntime-config.cmake

  Add the installation prefix of "onnxruntime" to CMAKE_PREFIX_PATH or set
  "onnxruntime_DIR" to a directory containing one of the above files.  If
  "onnxruntime" provides a separate development package or SDK, be sure it
  has been installed.


-- Downloading SIMDJSON source
-- Downloading SIMDJSON source -- done
-- WASI-NN: Build piper backend for WASI-NN
CMake Warning at cmake/WASINNDeps.cmake:238 (find_package):
  By not providing "Findonnxruntime.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "onnxruntime", but CMake did not find one.

  Could not find a package configuration file provided by "onnxruntime" with
  any of the following names:

    onnxruntimeConfig.cmake
    onnxruntime-config.cmake

  Add the installation prefix of "onnxruntime" to CMAKE_PREFIX_PATH or set
  "onnxruntime_DIR" to a directory containing one of the above files.  If
  "onnxruntime" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  cmake/WASINNDeps.cmake:185 (wasmedge_setup_piper)
  plugins/wasi_nn/CMakeLists.txt:231 (wasmedge_setup_wasinn_target)


-- Downloading piper source
-- Configuring incomplete, errors occurred!
```

### Screenshots

_No response_

### Any logs you want to share for showing the specific issue

_No response_

### Components

Others

### WasmEdge Version or Commit you used

master (41fb955ed9d245ba9c34fd274744df861e68a8e6)

### Operating system information

Ubuntu 20.04

### Hardware Architecture

x86_64

### Compiler flags and options

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 # SPDX-FileCopyrightText: 2019-2024 Second State INC
 
-cmake_minimum_required(VERSION 3.15)
+cmake_minimum_required(VERSION 3.18)
 cmake_policy(SET CMP0091 NEW)
 if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.24.0")
   cmake_policy(SET CMP0135 NEW)

--- a/examples/embed_cxx/CMakeLists.txt
+++ b/examples/embed_cxx/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.18)
 
 project(embed_cxx CXX)
 

--- a/utils/docker/Dockerfile.ubuntu-base
+++ b/utils/docker/Dockerfile.ubuntu-base
@@ -5,7 +5,6 @@ ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update && \
     apt-get upgrade -y && \
     apt-get install -y \
-        cmake \
         curl \
         dpkg-dev \
         g++ \
@@ -18,6 +17,9 @@ RUN apt-get update && \
 
 ### deps for ubuntu 20.04 ###
 FROM base AS deps-20
+
+RUN curl -sSf https://apt.kitware.com/kitware-archive.sh | sh
+RUN apt-get install -y cmake
 
 RUN apt-get install -y \
         llvm-12-dev \
@@ -33,6 +35,8 @@ ENV CXX=/usr/bin/clang++-12
 
 ### deps for ubuntu 22.04 ###
 FROM base AS deps-22
+
+RUN apt-get install -y cmake
 
 RUN apt-get install -y \
         llvm-15-dev \

